metron-platform/metron-storm-kafka/src/main/java/org/apache/metron/storm/kafka/flux/SimpleStormKafkaBuilder.java

```diff
@@ -0,0 +1,234 @@
+   */
+  public static class TupleBuilder<K, V> extends KafkaSpoutTupleBuilder<K,V> {
+    private List<FieldsConfiguration> configurations;
+    private TupleBuilder(String topic, List<FieldsConfiguration> configurations) {
+      super(topic);
+      this.configurations = configurations;
+    }
+
+    /**
+     * Builds a list of tuples using the ConsumerRecord specified as parameter
+     *
+     * @param consumerRecord whose contents are used to build tuples
+     * @return list of tuples
+     */
+    @Override
+    public List<Object> buildTuple(ConsumerRecord<K, V> consumerRecord) {
+      Values ret = new Values();
+      for(FieldsConfiguration config : configurations) {
+        ret.add(config.recordExtractor.apply(consumerRecord));
+      }
+      return ret;
+    }
+  }
+
+  private String topic;
+
+  /**
+   * Create an object with the specified properties.  This will expose fields "key" and "value."
+   * @param kafkaProps The special kafka properties
+   * @param topic The kafka topic. TODO: In the future, support multiple topics and regex patterns.
+   * @param zkQuorum The zookeeper quorum.  We will use this to pull the brokers from this.
+   */
+  public SimpleStormKafkaBuilder( Map<String, Object> kafkaProps
+                                , String topic
+                                , String zkQuorum
+                                )
+  {
+    this(kafkaProps, topic, zkQuorum, Arrays.asList("key", "value"));
+  }
+
+  /**
+   * Create an object with the specified properties and exposing the specified fields.
+   * @param kafkaProps The special kafka properties
+   * @param topic The kafka topic. TODO: In the future, support multiple topics and regex patterns.
+   * @param zkQuorum The zookeeper quorum.  We will use this to pull the brokers from this.
+   * @param fieldsConfiguration The fields to expose in the storm tuple emitted.
+   */
+  public SimpleStormKafkaBuilder( Map<String, Object> kafkaProps
+                                , String topic
+                                , String zkQuorum
+                                , List<String> fieldsConfiguration
+                                )
+  {
+    super( modifyKafkaProps(kafkaProps, zkQuorum)
+         , createStreams(fieldsConfiguration, topic)
+         , createTuplesBuilder(fieldsConfiguration, topic)
+         );
+    this.topic = topic;
+  }
+
+  /**
+   * Get the kafka topic.  TODO: In the future, support multiple topics and regex patterns.
+   * @return
+   */
+  public String getTopic() {
+    return topic;
+  }
+
+  /**
+   * Create a StormKafkaSpout from a given topic, zookeeper quorum and fields.  Also, configure the spout
+   * using a Map that configures both kafka as well as the spout (see the properties in SpoutConfiguration).
+   * @param topic
+   * @param zkQuorum
+   * @param fieldsConfiguration
+   * @param kafkaProps  The aforementioned map.
+   * @return
+   */
+  public static StormKafkaSpout create( String topic
```

Comment:
Shouldn't this and the stuff calling it all be generic? Why bother typing it, if we're just killing the types during the create()?

I didn't dig too much into it, but it seems like at least a few things can be typed.